File: editorial_ch/chap5-7/prob5-7-3.py
# 輸入
N = int(input())
A = list(map(int, input().split()))

# 將陣列 A 全部排序
A.sort()

# 求出答案
Answer = 0
for i in range(1, N + 1):
	Answer += A[i - 1] * (-N + 2 * i - 1)

# 輸出
print(Answer)

File: editorial_ch/chap5-7/prob5-7-3.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 輸入
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		long[] A = new long[N];
		for (int i = 1; i <= N; i++) {
			A[i - 1] = sc.nextLong();
		}
		
		// 將陣列 A 全部排序
		Arrays.sort(A);
		
		// 求出答案
		long Answer = 0;
		for (int i = 1; i <= N; i++) {
			Answer += A[i - 1] * (-N + 2L * i - 1L);
		}
		
		// 輸出
		System.out.println(Answer);
	}
}